pkg/pgsql/server/fmessages/bind.go

```diff
@@ -0,0 +1,168 @@
+	// The parameter format codes. Each must presently be zero (text) or one (binary).
+	ParameterFormatCodes []int16
+	// Array of the values of the parameters, in the format indicated by the associated format code. n is the above length.
+	ParamVals []interface{}
+	// The result-column format codes. Each must presently be zero (text) or one (binary).
+	ResultColumnFormatCodes []int16
+}
+
+func ParseBindMsg(payload []byte) (BindMsg, error) {
+	b := bytes.NewBuffer(payload)
+	r := bufio.NewReaderSize(b, len(payload))
+	destPortalName, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	preparedStatement, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	// The number of parameter format codes that follow (denoted C below).
+	// This can be zero to indicate that there are no parameters or that the parameters all use the default format (text);
+	// or one, in which case the specified format code is applied to all parameters; or it can equal the actual number
+	// of parameters.
+	parameterFormatCodeNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	if parameterFormatCodeNumber < 0 {
+		return BindMsg{}, pgserrors.ErrMalformedMessage
+	}
+	parameterFormatCodes := make([]int16, parameterFormatCodeNumber)
+	for k := 0; k < int(parameterFormatCodeNumber); k++ {
+		p, err := getNextInt16(r)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		parameterFormatCodes[k] = p
+	}
+	// The number of parameter values that follow (possibly zero). This must match the number of parameters needed by the query.
+	pCount, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+
+	// Handling format codes: see resultColumnFormatCodesNumber property comment
+	forceTXT := false
+	forceBIN := false
+	if len(parameterFormatCodes) == 0 {
+		forceTXT = true
+	}
+	if len(parameterFormatCodes) == 1 {
+		switch parameterFormatCodes[0] {
+		case 0:
+			forceTXT = true
+		case 1:
+			forceBIN = true
+		default:
+			return BindMsg{}, pgserrors.ErrMalformedMessage
+		}
+	}
+
+	if len(parameterFormatCodes) > 1 && len(parameterFormatCodes) != int(pCount) {
+		return BindMsg{}, pgserrors.ErrMalformedMessage
+	}
+	totalParamLen := 0
+	params := make([]interface{}, 0)
+	for i := 0; i < int(pCount); i++ {
+		pLen, err := getNextInt32(r)
+		if pLen < 0 {
+			return BindMsg{}, pgserrors.ErrNegativeParameterValueLen
+		}
+		if err != nil {
+			return BindMsg{}, err
+		}
+		totalParamLen += int(pLen)
+		if totalParamLen > math.MaxInt32 {
+			return BindMsg{}, pgserrors.ErrParametersValueSizeTooLarge
+		}
+		pVal := make([]byte, pLen)
+		_, err = r.Read(pVal)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		if forceTXT {
+			params = append(params, string(pVal))
+			continue
+		}
+		if forceBIN {
+			params = append(params, pVal)
+			continue
+		}
+
+		switch parameterFormatCodes[i] {
+		case 0:
+			params = append(params, string(pVal))
+		case 1:
+			params = append(params, pVal)
+		}
```

Comment:
This got lost somewhere along other comments, a check similar to the one on  [line 98](https://github.com/codenotary/immudb/pull/840/files#diff-7841e1c8241c390dd964fe42839fca3f6b0023e74dee5052b57cf34fcb5d1d80R98) is needed here too:

```suggestion
		default:
			return BindMsg{}, pgserrors.ErrMalformedMessage
		}
```